src/leap/keymanager/__init__.py

```diff
@@ -219,13 +220,21 @@ def _fetch_keys_from_server(self, address):
             res = self._get(self._nickserver_uri, {'address': address})
             res.raise_for_status()
             server_keys = res.json()
+
             # insert keys in local database
             if self.OPENPGP_KEY in server_keys:
+                # nicknym server is authoritative for its own domain,
+                # for other domains the key might come from key servers.
+                validation_level = ValidationLevel.Weak_Chain
+                if (address.split("@")[1] ==
+                        urlparse(self._nickserver_uri).hostname):
```

Comment:
this is perfectly valid and understood, but maybe factoring out a couple of helper functions out of here the check can be made more readable:
```
username, domain = split_full_userid(address)
if domain == get_domain(self._nickserver_uri):
 ...
```
what do you think?

Also, maybe it's a good thing to validate for well-formed address (ie, only one @, non-null characters), and non-null nickserver_uri here? Or maybe i'm being overly paranoid.